src/components/SelectionList/BaseListItem.js

```diff
@@ -33,6 +34,14 @@ function BaseListItem({
     const isUserItem = lodashGet(item, 'icons.length', 0) > 0;
     const ListItem = isUserItem ? UserListItem : RadioListItem;
 
+    const rightHandSideComponentRender = () => {
+        if (canSelectMultiple || !rightHandSideComponent) {
+            return null; // no right hand side component
+        }
+
+        return typeof rightHandSideComponent === 'function' ? rightHandSideComponent(item) : rightHandSideComponent;
```

Comment:
Let's use the `if` block here.
```suggestion
        if (typeof rightHandSideComponent === 'function' ) {
            return rightHandSideComponent(item)
        }
        
        return rightHandSideComponent;
```

Reply:
I'm doing it only for you!